### PR TITLE
Add field name to column name if title not given

### LINKED ISSUE #9871: [BUG] TableColumn doesn't use `field` when `title` is not set
#### ALL software version info (bokeh, python, notebook, OS, browser, any other relevant packages)
Bokeh 2.0.1

#### Description of expected behavior and the observed behavior
The docstring of `TableColumn.title` says: "If not set, column's data field is used instead."
But this is not the case, the title becomes `null` instead.

#### Complete, minimal, self-contained example code that reproduces the issue

```python
from bokeh.io import show
from bokeh.models import DataTable, TableColumn, ColumnDataSource

cds = ColumnDataSource(dict(value=[1, 2, 3]))
dt = DataTable(columns=[TableColumn(field='value')], source=cds)
show(dt)
```

### PR DESCRIPTION
**Pull request by ayushiitr2901**
I thought I should just go ahead and make the change. This most probably will work fine as the library would use column field name instead if the title is not externally given.
- [ ] issues: fixes #9871

### CODE REVIEW

File: bokehjs/src/lib/models/widgets/tables/table_column.ts [hunk 1]
```diff
@@ -49,7 +49,7 @@ export class TableColumn extends Model {
     return {
       id: uniqueId(),
       field: this.field,
-      name: this.title,
+      name: this.title || this.field,
```
Comment (mattpap) [inline bokehjs/src/lib/models/widgets/tables/table_column.ts]: Use new ES2020 operator `??`.

**Comment by mattpap** [comment]
@ayushiitr2901, you need to maintain labels, milestones, etc. on your issues and PRs. Follow https://github.com/bokeh/bokeh/wiki/BEP-1:-Issues-and-PRs-management. In this case `status: ready` is sufficient (if you are done with changes, otherwise `status: WIP`).

File: bokehjs/src/lib/models/widgets/tables/table_column.ts (hunk 1 above)
Comment (ayushiitr2901) [reply bokehjs/src/lib/models/widgets/tables/table_column.ts]: Woah, I did not know about these js updates. I will update the code accordingly.

**Comment by ayushiitr2901** [comment]
@mattpap I don't think I have the edit rights to add a label to an open Pull Request. Should I use the Issues API for a PR as well or is there a way I can't find it. Could you please help me with the same.

**Comment by mattpap** [comment]
I was pretty sure that you should be allowed to label PRs you are an author of, regardless of your given permissions. I checked on my contributions to other projects and apparently that's not the case. That's pretty disappointing. We will give you the right permissions, so that you can maintain your work on your own in future. For now I will do the labeling.

**Comment by ayushiitr2901** [comment]
That sounds great. Thanks

**Comment by bryevdv** [comment]
@mattpap I went ahead and sent the invite to join @bokeh/dev @ayushiitr2901 apologies I should have thought to invite you  earlier

**Comment by ayushiitr2901** [comment]
Thanks, I have successfully joined Bokeh/Dev.

**Pull request merged**